src/invidious/users.cr

```diff
@@ -3,6 +3,24 @@ require "crypto/bcrypt/password"
 # Materialized views may not be defined using bound parameters (`$1` as used elsewhere)
 MATERIALIZED_VIEW_SQL = ->(email : String) { "SELECT cv.* FROM channel_videos cv WHERE EXISTS (SELECT subscriptions FROM users u WHERE cv.ucid = ANY (u.subscriptions) AND u.email = E'#{email.gsub({'\'' => "\\'", '\\' => "\\\\"})}') ORDER BY published DESC" }
 
+def create_user(sid, email)
+  token = Base64.urlsafe_encode(Random::Secure.random_bytes(32))
+
+  user = Invidious::User.new({
+    updated:           Time.utc,
+    notifications:     [] of String,
+    subscriptions:     [] of String,
+    email:             email,
+    preferences:       Preferences.new(CONFIG.default_user_preferences.to_tuple),
+    password:          nil,
+    token:             token,
+    watched:           [] of String,
+    feed_needs_update: true,
+  })
+
+  return user, sid
+end
```

Comment:
The problem I see with an empty password is that you can'tto differentiate between two users with the same name but using two different authentication methods/providers (that could result in an account hijack).

I'd suggest using the password field to store the account type (like `oauth:github.com`), or to add a dedicated database field (and add a migration, which is a much cleaner solution).

Reply:
I see 3 scenarios: 
- User can use multiple services to get into his account (its a feature)
- User wants to hijack any account via oauth as it doesn't check/require password (its bad?). The administrator/host should select oauth providers that do not conflict usernames or where email address ownership is verified.
- User wants to hijack an oauth account via an internal account (safe), the password is nil and he cannot log into it.

The solutions are the follows
- Adding a field to the db: adds complexity and requires migration, might break the first scenario.
- Adding it to the password 
- Adding suffix to usernames to indicate auth_type  ("test@github", "test@internal",...), might also require migrations

I'd favor either the password or the username approach